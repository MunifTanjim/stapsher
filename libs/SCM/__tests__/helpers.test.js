const helpers = require('../../../__tests__/helpers')

helpers.addSnapshotSerializers()

const { StapsherError } = require('../../Error')

const { parseFile } = require('../helpers')

describe('libs/SCM/helpers', () => {
  describe('parseFile', () => {
    let jsonBlob = JSON.stringify(helpers.readConfigFile())
    let yamlBlob = helpers.getSampleConfig()
    let parsedFile = helpers.readConfigFile()

    it('parses json blob', () => {
      expect(parseFile(jsonBlob, 'json')).toEqual(parsedFile)
    })

    it('parses yaml blob', () => {
      expect(parseFile(yamlBlob, 'yaml')).toEqual(parsedFile)
      expect(parseFile(yamlBlob, 'yml')).toEqual(parsedFile)
    })

    it('throws error if extension not supported', () => {
      try {
        parseFile(jsonBlob, 'snap')
      } catch (err) {
        expect(err).toBeInstanceOf(StapsherError)
        expect(err).toMatchSnapshot()
      }
    })

    it('throws error if parse fails', () => {
      try {
        parseFile(yamlBlob, 'json')
      } catch (err) {
        expect(err).toBeInstanceOf(StapsherError)
        expect(err).toMatchSnapshot()
      }
    })
  })
})
