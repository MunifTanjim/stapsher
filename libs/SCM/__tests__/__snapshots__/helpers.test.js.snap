// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`libs/SCM/helpers parseFile throws error if extension not supported 1`] = `
{
  "code": "UNSUPPORTED_EXTENSION",
  "statusCode": 422,
  "cause": {
    "extension": "snap"
  }
}
`;

exports[`libs/SCM/helpers parseFile throws error if parse fails 1`] = `
{
  "code": "FILE_PARSE_FAILED",
  "statusCode": 422,
  "cause": "SyntaxError: Unexpected token c in JSON at position 1"
}
`;
